### DISCUSSION TITLE
Finding delegate libraries in nonstandard locations?

### DISCUSSION THREAD

**Discussion by blmatthews** [ImageMagick/ImageMagick#2355] [Help]
I'm building ImageMagick from source but can't get it to recognize that I have the delegate libraries I want installed but in a non-standard location. I'm on macOS Mojave and have the standard tools installed and working. If I configure ImageMagick thusly:

`./configure --prefix=$HOME/installed --with-perl`

It makes, checks, installs, and works just fine, except without PNG support. The configure doesn't recognize I have libpng installed. configure outputs:

`  PNG               --with-png=yes		no`

libpng is installed under $HOME/installed:

```
$ ls -ln $HOME/installed/lib/libpng*
lrwxr-xr-x  1 501  20       10 Aug  2 14:17 /Users/blm/installed/lib/libpng.a -> libpng16.a
lrwxr-xr-x  1 501  20       14 Aug  2 14:17 /Users/blm/installed/lib/libpng.dylib -> libpng16.dylib
lrwxr-xr-x  1 501  20       11 Aug  2 14:17 /Users/blm/installed/lib/libpng.la -> libpng16.la
-rwxr-xr-x  1 501  20   253708 Aug  2 14:17 /Users/blm/installed/lib/libpng16.16.dylib
-rw-r--r--  1 501  20  1024496 Aug  2 14:17 /Users/blm/installed/lib/libpng16.a
lrwxr-xr-x  1 501  20       17 Aug  2 14:17 /Users/blm/installed/lib/libpng16.dylib -> libpng16.16.dylib
-rwxr-xr-x  1 501  20      933 Aug  2 14:17 /Users/blm/installed/lib/libpng16.la
```
I've tried various combinations of specifying LDFLAGS and PNG_LIBS on the configure command line, but nothing I've done gets it to recognize I have libpng installed.

Thanks for any help or direction you can provide!

**Comment by fmw42** [comment]
Installing from source does not provide any delegates. You need to install any delegates you want before installing ImageMagick.  I would suggest you install from a binary. See https://imagemagick.org/script/download.php#macosx. Easiest in to use Homebrew.  

However if you still want to install from source, here is how I do it on my Mac by installing all my delegates using MacPorts and then install ImageMagick from source.  See https://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=29100

**Reply by blmatthews** [reply]
Yes, I realize it doesn't provide any delegates by default. I've installed the library for the one delegate I want right now (PNG), my problem is getting the ImageMagick configure to recognize that I've installed that library. I tried the various configure command lines in your second link, but none of them made configure recognize I have libpng installed.

**Comment by fmw42** [comment]
If libpng is installed properly, ImageMagick will show it with convert -version.  So I suspect you have not installed it properly. Check the install log for libpng and see if there are any errors. I think it needs some other delegate library to  be installed before installing libpng. Check the libpng web site to see what delegates it needs.

**Reply by blmatthews** [reply]
When I install libpng, make, make test, and make install all work without error, and as a test I installed the perl module Image::PNG::Libpng (which is, unfortunately too low-level or I'd just use that), which uses libpng, and that makes, tests, and installs just fine, so I'm fairly confident libpng is installed properly and working.

On convert -version showing it or not, as I said, configure doesn't find it, so of course an executable built by a Makefile generated by configure isn't going to know about it. The question remains, how do I make configure recognize I have libpng installed?

**Reply by fmw42** [reply]
try putting the path to libpng in your $PATH environment variable and then install ImageMagick again.

**Reply by blmatthews** [reply]
I would have been surprised if that worked, that's not what PATH is for, but I tried it, and no change.

**Comment by snibgo** [comment]
Perhaps you need "autoconf" before "./configure".

**Reply by blmatthews** [reply]
Thanks, tried that. The generated configure had two trivial differences from the configure included in the source download, and running the generated configure still didn't find libpng.
